kura/org.eclipse.kura.container.orchestration.provider/src/main/java/org/eclipse/kura/container/orchestration/provider/impl/DockerServiceImpl.java

```diff
@@ -120,93 +118,70 @@ public void updated(Map<String, Object> properties) {
+                .setInternalPort(parseInternalPortsFromDockerPs(container.getPorts()))
+                .setExternalPort(parseExternalPortsFromDockerPs(container.getPorts()))
+                .setContainerState(convertDockerStateToFrameworkState(container.getState()))
+                .setFrameworkManaged(isFrameworkManaged(container)).build()));
 
-        int temp = 0;
-        for (Container cont : containers) {
-            avaliableContainers[temp] = cont.getId();
-            temp++;
-        }
-
-        return avaliableContainers;
+        return result;
 
     }
 
-    @Override
-    public ContainerDescriptor[] listByContainerDescriptor() {
-        ContainerDescriptor[] avaliableContainers;
-        if (!testConnection()) {
-            throw new IllegalStateException(UNABLE_TO_CONNECT_TO_DOCKER_CLI);
-        }
-
-        List<Container> containers = this.dockerClient.listContainersCmd().withShowAll(true).exec();
-
-        avaliableContainers = new ContainerDescriptor[containers.size()];
-
-        if (!containers.isEmpty()) {
-            int temp = 0;
-            for (Container cont : containers) {
-
-                String tag = cont.getImage().split(":")[0];
-                String version = "";
+    private Boolean isFrameworkManaged(Container container) {
+        String containerName = getContainerName(container);
+        return this.dockerServiceListeners.values().stream()
+                .anyMatch(registeredContainerName -> registeredContainerName.equals(containerName));
+    }
 
-                if (cont.getImage().split(":").length > 1) {
-                    version = cont.getImage().split(":")[1];
-                }
+    private String getContainerName(Container container) {
+        return container.getNames()[0].replace("/", "");
+    }
 
-                avaliableContainers[temp] = ContainerDescriptor.builder()
-                        .setContainerName(cont.getNames()[0].replace("/", "")).setContainerImage(tag)
-                        .setContainerImageTag(version).setContainerID(cont.getId())
-                        .setInternalPort(parseInternalPortsFromDockerPs(cont.getPorts()))
-                        .setExternalPort(parseExternalPortsFromDockerPs(cont.getPorts()))
-                        .setContainerState(convertDockerStateToESFState(cont.getState())).setIsEsfManaged(false)
-                        .build();
-                temp++;
-            }
+    private String getContainerVersion(Container container) {
+        String version = "";
+        if (container.getImage().split(":").length > 1) {
+            version = container.getImage().split(":")[1];
         }
-        return avaliableContainers;
+        return version;
+    }
 
+    private String getContainerTag(Container container) {
+        return container.getImage().split(":")[0];
     }
 
     private int[] parseExternalPortsFromDockerPs(ContainerPort[] ports) {
```

Comment:
This method will always return an empty array, the same applies to parseInternalPortsFromDockerPs, ArrayUtils.add does not modify the passed array argument, which is immutable.

Reply:
The `ipFormatTest.equals("::")` check fails on my device, the reported id in that case is `0.0.0.0`.
Should we expose the IP in our APIs?